packages/web-components/src/components/fade-in-out/fade-in-out.ts

```diff
@@ -0,0 +1,256 @@
+ *
+ * @example
+ * import '@carbon/ibmdotcom-styles/scss/internal/scroll-into-view/_scroll-into-view.scss';
+ * import '@carbon/ibmdotcom-web-components/es/components/fade-in-out/fade-in-out.js';
+ *
+ * As an example, the function can be called to target all instances of the
+ * elements in a list:
+ *
+ * const list = 'bx--content-block, bx--content-group';
+ *
+ * For default values of 400ms and 'one and done' play:
+ * <dds-fade-in-out selectorTargets="${selectorTargets}"/>
+ *
+ * With 'continuous play' option:
+ * <dds-fade-in-out selectorTargets="${selectorTargets}" iterations="true"/>
+ *
+ * For custom delay time, set within targeted class in the application's CSS code as such:
+ *
+ * .bx--content-block {
+ *   --#{$dds-prefix}--fade-in-out-delay: 250ms;
+ * }
+ *
+ * @element dds-fade-in-out
+ */
+@customElement(`${ddsPrefix}-fade-in-out`)
+class DDSFadeInOut extends StableSelectorMixin(LitElement) {
+  /**
+   * Intersection Observer that watches outer viewport.
+   *
+   * @private
+   */
+  private _rootObserver: IntersectionObserver | null = null;
+
+  /**
+   * Intersection observer that watches the inner viewport.
+   *
+   * @private
+   */
+  private _innerObserver: IntersectionObserver | null = null;
+
+  /**
+   * Resize observer to trigger rootMargin recalculations
+   *
+   * @private
+   */
+  private _resizeObserver: any | null = null;
+
+  /**
+   * Root margin option for inner observer.
+   *
+   * @private
+   */
+  private _rootMargin?: string;
+
+  /**
+   * The inner viewport calculation for the root margins.
+   *
+   * This calculation is done to retrieve the best fitting top and bottom
+   * margin for the fade animation to trigger/remove from elements in a
+   * user's screen.
+   *
+   * The resulting value is the optimal point where a user's attention will be
+   * grabbed by the animation without restricting their view and perception of
+   * the adopting website. The displayed elements will keep the user's attention
+   * for a longer time as they scroll down the website.
+   *
+   * @private
+   */
+  // eslint-disable-next-line class-methods-use-this
+  private _getViewportMargin() {
+    return `-${((document.documentElement.clientHeight * _colSpan) / breakpoints.max.columns).toString()}px 0px`;
+  }
+
+  /**
+   * Cleans observers upon update or exit, and creates new instances if needed.
+   *
+   * @param [options] The options.
+   * @param [options.create] `true` to create the new intersection observer.
+   */
+  private _cleanAndCreateObservers({ create }: { create?: boolean } = {}) {
+    this._cleanAndCreateRootObserver();
+    this._cleanAndCreateInnerObserver();
+    this._cleanAndCreateResizeObserver();
+
+    if (create) {
+      this._cleanAndCreateRootObserver({ create: true });
+      this._cleanAndCreateResizeObserver({ create: true });
+
+      const { selectorTargets } = this;
+      if (selectorTargets) {
+        forEach(this.ownerDocument!.querySelectorAll(selectorTargets), item => {
```

Comment:
I may have made a wrong suggestion here, my apologies if it's the case. The `selectorTargets` search should be scoped within `<dds-fade-in-out>`:

```suggestion
        forEach(this.querySelectorAll(selectorTargets), item => {
```